src/System.Runtime.Extensions/src/System/IO/StreamWriter.cs

```diff
@@ -1024,8 +1046,25 @@ private bool HaveWrittenPreamble_Prop
             if (flushStream)
             {
                 await stream.FlushAsync().ConfigureAwait(false);
+                _this.ReturnCharBuffer();
             }
         }
         #endregion
+
+        private char[] CharBuffer => _rentedCharBuffer ?? RentCharBuffer();
```

Comment:
You've confirmed this inlines?

Reply:
Added Agressive to be sure, would depend on function chain